### PR TITLE
Checking for image URL

### PR DESCRIPTION
**Pull request by EugeneElkin**

### CODE REVIEW

File: src/validationHelper.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+
+module powerbi.extensibility.utils.dataview {
+    export module validationHelper {
+        export function checkIsImageUrlAllowable(url: string): boolean {
+            // Excludes all URLs that don't contain .gif .jpg .png or .svg extensions.
+            // Also excludes directives "javascript:" and "data:".
+            return (/\.(gif|jpg|png|svg)$/i).test(url) && !(/(javascript:|data:)/i).test(url);
```
Comment (ignatvilesov) [inline src/validationHelper.ts]: Looks like this RegExp does'n support base64 images.

File: test/validationHelperTest.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        it("URL javascript: directive checking", () => {
+            expect(validationHelper.checkIsImageUrlAllowable("jAvAscrIpt:alert('XSS');")).toBe(false);
+            expect(validationHelper.checkIsImageUrlAllowable("jAvAscrIpt:alert('XSS');.png")).toBe(false);
+            expect(validationHelper.checkIsImageUrlAllowable("jaascript:alert('XSS');.png")).toBe(true);
+        });
+
+        it("URL data: directive checking", () => {
+            expect(validationHelper.checkIsImageUrlAllowable("data:image/jpeg;base64,/9j/4AAQSkZJRgABAQEASABIAAD")).toBe(false);
+            expect(validationHelper.checkIsImageUrlAllowable("data:image/jpeg;base64,/9j/4AAQSkZJRgABAQEASABIAAD.png")).toBe(false);
+            expect(validationHelper.checkIsImageUrlAllowable("dat:image/jpeg;base64,/9j/4AAQSkZJRgABAQEASABIAAD.png")).toBe(true);
```
Comment (ignatvilesov) [inline test/validationHelperTest.ts]: This base64 string isn't valid due to specifying extension at the end of string

File: src/validationHelper.ts (hunk 1 above)
Comment (EugeneElkin) [reply src/validationHelper.ts]: That's true. I deliberately excluded base64 images

File: lib/index.d.ts [hunk 3]
```diff
@@ -61,6 +61,11 @@ declare module powerbi.extensibility.utils.dataview {
     }
 }
 declare module powerbi.extensibility.utils.dataview {
+    module validationHelper {
+        function checkIsImageUrlAllowable(url: string): boolean;
```
Comment (AviSander) [inline lib/index.d.ts]: rename: isImageUrlAllowed()

File: src/validationHelper.ts (hunk 1 above)
Comment (AviSander) [inline src/validationHelper.ts]: please wait for Assaf to signoff as well

File: lib/index.d.ts (hunk 3 above)
Comment (EugeneElkin) [reply lib/index.d.ts]: Got it. I will do.

**Pull request merged**